### PR TITLE
2502 lesmis map fixes

### PR DESCRIPTION
**Pull request by tcaiger**
### Changes:

See annotation of files for a description of the changes.

---

### Screenshots:
See issue

### CODE REVIEW

File: packages/lesmis/src/api/queries/useMapOverlayReportData.js [hunk 1]
```diff
@@ -78,20 +79,24 @@ const processMeasureData = ({
...
         serieses,
         hiddenValues,
+        radiusScaleFactor,
       );
 
       return {
         ...entity,
         ...measure,
         isHidden,
+        radius,
```
Comment (tcaiger) [inline packages/lesmis/src/api/queries/useMapOverlayReportData.js:99]: Re-add support for radius map overlays

File: packages/ui-components/src/components/Map/Legend/SpectrumLegend.js [hunk 2]
```diff
@@ -111,8 +120,10 @@ export const SpectrumLegend = React.memo(({ series }) => {
         <LegendEntry
           marker={getMarkerForOption(LEGEND_SHADING_ICON, noDataColour)}
           label="No data"
+          dataKey={dataKey}
+          hiddenValues={hiddenValues}
+          onClick={setValueHidden}
```
Comment (tcaiger) [inline packages/ui-components/src/components/Map/Legend/SpectrumLegend.js:125]: Fix spectrum legend no data interaction

File: packages/ui-components/src/components/Map/MarkerLayer.js [hunk 3]
```diff
@@ -19,13 +19,9 @@ const ShadedPolygon = styled(Polygon)`
   }
 `;
 
-const getText = (measure, serieses) => {
-  const { name } = measure;
+const getText = ({ name, ...measure }, serieses) => {
```
Comment (tcaiger) [inline packages/ui-components/src/components/Map/MarkerLayer.js:22]: remove the name key from the measure data as it's not expected in `getSingleFormattedValue`

File: packages/ui-components/src/components/Map/Markers/CircleProportionMarker.js [hunk 4]
```diff
@@ -16,39 +16,35 @@ const HoverCircle = styled(CircleMarker)`
   }
 `;
 
-export const CircleProportionMarker = React.memo(
```
Comment (tcaiger) [inline packages/ui-components/src/components/Map/Markers/CircleProportionMarker.js:19]: Remove the radius scale handling here as it is handled when the data is initially parsed now

File: packages/ui-components/src/components/Map/Markers/MeasurePopup.js [hunk 5]
```diff
@@ -9,9 +9,9 @@ import { getFormattedInfo } from '../utils';
 import { PopupMarker } from './PopupMarker';
 
 const buildHeaderText = (measureData, popupHeaderFormat) => {
-  const { organisationUnitCode, name } = measureData;
+  const { code, name } = measureData;
   const replacements = {
-    code: organisationUnitCode,
```
Comment (tcaiger) [inline packages/ui-components/src/components/Map/Markers/MeasurePopup.js:14]: Fix missed update on refactor of `organisationUnitCode`  to `code` to entity server

File: packages/ui-components/src/components/Map/MarkerLayer.js [hunk 6]
```diff
@@ -19,13 +19,9 @@ const ShadedPolygon = styled(Polygon)`
...
   const hasMeasureValue = measure || measure === 0;
```
Comment (edmofro) [inline packages/ui-components/src/components/Map/MarkerLayer.js:23]: I don't understand how `measure === 0` could be true here with the destructuring that's happening

**Review by edmofro — CHANGES_REQUESTED** [review]
Mostly looks good, just one question on a line that didn't actually change but maybe should

File: packages/ui-components/src/components/Map/MarkerLayer.js (hunk 6 above)
Comment (tcaiger) [reply packages/ui-components/src/components/Map/MarkerLayer.js]: I copied that across from tupaia to try and not miss any edge cases but it definitely doesn't make sense anymore. I tried debugging it on web-frontend to figure out what it's there for but couldn't find any examples of when it's needed.

**Review by edmofro — APPROVED** [review]

**Pull request merged**